utils/directories.cc

```diff
@@ -151,6 +151,17 @@ future<> directories::verify_owner_and_mode(fs::path path, recursive recursive)
     return do_verify_owner_and_mode(std::move(path), recursive, 0);
 }
 
+seastar::future<> directories::create_and_verify_directories() {
+    // Note: creation of hints_dir and view_hints_dir is
+    //       responsibility of db::hints::directory_initializer.
+    utils::directories::set dir_set;
```

Comment:
Does the directories::set become internal class for utils/directories.cc after this change?

Reply:
No, it does not. It is still used by:
`db::hints::directory_initializer::impl::ensure_created_and_verified()`

Reply:
OK, but it now looks like

```
        utils::directories::set dir_set;
        dir_set.add_sharded(_hints_directory);

        manager_logger.debug("Creating and validating hint directories: {}", _hints_directory);
        co_await _dirs.create_and_verify(std::move(dir_set));
```

just a single path. Worth encapsulating this class out of global scope

Reply:
Okay, fixed. I made `utils::directories::set` an internal type and added a public method called `create_and_verify_sharded_directory()` that is used by `directory_initializer`.